plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/SyncPartitionMetadataProcedure.java

```diff
@@ -137,23 +139,46 @@ private void doSyncPartitionMetadata(ConnectorSession session, ConnectorAccessCo
                 accessControl.checkCanDeleteFromTable(null, new SchemaTableName(schemaName, tableName));
             }
 
-            Location tableLocation = Location.of(table.getStorage().getLocation());
-
-            Set<String> partitionsInMetastore = metastore.getPartitionNames(schemaName, tableName)
+            Set<String> canonicalPartitionNamesInMetastore = metastore.getPartitionNames(schemaName, tableName)
                     .map(ImmutableSet::copyOf)
                     .orElseThrow(() -> new TableNotFoundException(schemaTableName));
-            Set<String> partitionsInFileSystem = listPartitions(fileSystemFactory.create(session), tableLocation, table.getPartitionColumns(), caseSensitive);
+            String tableStorageLocation = table.getStorage().getLocation();
+            String tableLocationDirectory = tableStorageLocation.endsWith("/") ? tableStorageLocation : tableStorageLocation + "/";
+            Set<String> actualPartitionNamesInMetastore = metastore.getPartitionsByNames(schemaName, tableName, canonicalPartitionNamesInMetastore.stream().collect(toImmutableList()))
```

Comment:
No need for a stream:
`ImmutableList.copyOf(canonicalPartitionNamesInMetastore)`